### PR TITLE
Makes it so PDAs can be used when laying down

### PR DESCRIPTION
**Pull request by Seris02**
## About The Pull Request

Makes it so PDAs can be used when laying down. (Or at least the messaging and other buttons parts of it, IDs, pens and flashlights on the PDA still need to be standing up to use)

## Why It's Good For The Game

Because it's annoying to have to stand up to PDA someone when I'm cuddling someone else.

## Changelog
:cl:
tweak: PDAs can be used when laying down.
/:cl:

### CODE REVIEW

**Comment by xXPawnStarrXx** [comment]
It's hell trying to PDA while lewding. 10/10

**Comment by SpaceVampire** [comment]
based

File: code/game/objects/items/devices/PDA/PDA.dm [hunk 1]
```diff
@@ -437,8 +437,9 @@ GLOBAL_LIST_EMPTY(PDAs)
 	..()
 	var/mob/living/U = usr
 	//Looking for master was kind of pointless since PDAs don't appear to have one.
-
-	if(usr.canUseTopic(src, BE_CLOSE, FALSE, NO_TK) && !href_list["close"])
+	//SKYRAT EDIT CHANGE BEGIN
```
Comment (atakiya) [inline code/game/objects/items/devices/PDA/PDA.dm:440]:
```suggestion
	//SKYRAT EDIT CHANGE BEGIN
	//if(usr.canUseTopic(src, BE_CLOSE, FALSE, NO_TK) && !href_list["close"]) - SKYRAT EDIT - ORIGINAL
```

File: code/game/objects/items/devices/PDA/PDA.dm [hunk 2]
```diff
@@ -712,7 +713,9 @@ GLOBAL_LIST_EMPTY(PDAs)
 	var/t = stripped_input(U, "Please enter message", name)
 	if (!t || toff)
 		return
-	if(!U.canUseTopic(src, BE_CLOSE))
+	//SKYRAT EDIT CHANGE BEGIN
```
Comment (atakiya) [inline code/game/objects/items/devices/PDA/PDA.dm:717]:
```suggestion
	//SKYRAT EDIT CHANGE BEGIN
	//if(!U.canUseTopic(src, BE_CLOSE)) - SKYRAT EDIT - ORIGINAL
```

**Review by atakiya — CHANGES_REQUESTED** [review]
Hello! Here is a friendly reminder to [read this section of the modularization guide](https://github.com/Skyrat-SS13/Skyrat-tg/blob/master/modular_skyrat/readme.md#non-modular-changes-to-the-core-code---important).
I've gone ahead and suggested the appropiate edits.

Also what is with the random resorting of the two lines in the `.dme`?

File: code/game/objects/items/devices/PDA/PDA.dm (hunk 2 above)
Comment (Seris02) [reply code/game/objects/items/devices/PDA/PDA.dm:716]: Oh, sorry, I didn't actually see that last bit at the bottom about changes, I think my eyes just kind of skipped over it. Also I got no clue, the .dme does its own thing most of the time to be honest.

**Comment by Seris02** [comment]
Sorry, I didn't actually see that last bit at the bottom about changes, I think my eyes just kind of skipped over it. Also I got no clue, the .dme does its own thing most of the time to be honest.

**Comment by atakiya** [comment]
> 
> 
> Sorry, I didn't actually see that last bit at the bottom about changes, I think my eyes just kind of skipped over it.

No worries, just keep it in mind :)

> 
> 
> Also I got no clue, the .dme does its own thing most of the time to be honest.

General rule of thumb, if you're not adding/creating or removing/deleting any `.dm` files, you shouldn't save any changes in the `.dme`

I think I can fix that from my side though, not that it's a big issue as it just swapped some line locations, but usually it's better to not touch files when they don't need to be touched ^^

**Comment by Gandalf2k15** [comment]
This PR creates a lot of balance issues and concerns with IC interaction. While this may sound great in principle, it is not in practice. Sorry.

**Comment by Azarak** [comment]
> 
> 
> This PR creates a lot of balance issues and concerns with IC interaction. While this may sound great in principle, it is not in practice. Sorry.

Can you give examples?

**Comment by Gandalf2k15** [comment]
> > This PR creates a lot of balance issues and concerns with IC interaction. While this may sound great in principle, it is not in practice. Sorry.
> 
> Can you give examples?

You have been stunned but you are not stam-critted, you are able to use your PDA to message. You are laying down in a crate, hiding, you can use your PDA to stealth message. You are under flaps, you can use your PDA to stealth message, no radio required. You are buckled to a bed with your PDA on belt, you can use it to message with no sounds. These are a few, the list goes on. I also dislike that you don't need to put any effort into messaging someone else if you are laying down. Use the radio. PDA's can also be used to call bots, if you are down, you can call bots.

**Comment by Azarak** [comment]
> 
> 
> > > This PR creates a lot of balance issues and concerns with IC interaction. While this may sound great in principle, it is not in practice. Sorry.
> > 
> > 
> > Can you give examples?
> 
> You have been stunned but you are not stam-critted, you are able to use your PDA to message. You are laying down in a crate, hiding, you can use your PDA to stealth message. You are under flaps, you can use your PDA to stealth message, no radio required. You are buckled to a bed with your PDA on belt, you can use it to message with no sounds. These are a few, the list goes on. I also dislike that you don't need to put any effort into messaging someone else if you are laying down. Use the radio. PDA's can also be used to call bots, if you are down, you can call bots.

You can stand up on flaps, you can stand up inside a crate. You can make it so it requires a MOBILITY_USE and MOBILITY_STAND flags if you wanna prevent stunned or paralyzed people from using it

**Comment by Gandalf2k15** [comment]
> > > > This PR creates a lot of balance issues and concerns with IC interaction. While this may sound great in principle, it is not in practice. Sorry.
> > > 
> > > 
> > > Can you give examples?
> > 
> > 
> > You have been stunned but you are not stam-critted, you are able to use your PDA to message. You are laying down in a crate, hiding, you can use your PDA to stealth message. You are under flaps, you can use your PDA to stealth message, no radio required. You are buckled to a bed with your PDA on belt, you can use it to message with no sounds. These are a few, the list goes on. I also dislike that you don't need to put any effort into messaging someone else if you are laying down. Use the radio. PDA's can also be used to call bots, if you are down, you can call bots.
> 
> You can stand up on flaps, you can stand up inside a crate. You can make it so it requires a MOBILITY_USE and MOBILITY_STAND flags if you wanna prevent stunned or paralyzed people from using it

There's also the interaction issue for some players. I am afraid this PR isn't a good change at all.

**Comment by Azarak** [comment]
> 
> 
> > > > > This PR creates a lot of balance issues and concerns with IC interaction. While this may sound great in principle, it is not in practice. Sorry.
> > > > 
> > > > 
> > > > Can you give examples?
> > > 
> > > 
> > > You have been stunned but you are not stam-critted, you are able to use your PDA to message. You are laying down in a crate, hiding, you can use your PDA to stealth message. You are under flaps, you can use your PDA to stealth message, no radio required. You are buckled to a bed with your PDA on belt, you can use it to message with no sounds. These are a few, the list goes on. I also dislike that you don't need to put any effort into messaging someone else if you are laying down. Use the radio. PDA's can also be used to call bots, if you are down, you can call bots.
> > 
> > 
> > You can stand up on flaps, you can stand up inside a crate. You can make it so it requires a MOBILITY_USE and MOBILITY_STAND flags if you wanna prevent stunned or paralyzed people from using it
> 
> There's also the interaction issue for some players. I am afraid this PR isn't a good change at all.

Interaction issue? Standing up from laying on a couch to PDA then laying down again? How does that relate

**Comment by Gandalf2k15** [comment]
> > > > > > This PR creates a lot of balance issues and concerns with IC interaction. While this may sound great in principle, it is not in practice. Sorry.
> > > > > 
> > > > > 
> > > > > Can you give examples?
> > > > 
> > > > 
> > > > You have been stunned but you are not stam-critted, you are able to use your PDA to message. You are laying down in a crate, hiding, you can use your PDA to stealth message. You are under flaps, you can use your PDA to stealth message, no radio required. You are buckled to a bed with your PDA on belt, you can use it to message with no sounds. These are a few, the list goes on. I also dislike that you don't need to put any effort into messaging someone else if you are laying down. Use the radio. PDA's can also be used to call bots, if you are down, you can call bots.
> > > 
> > > 
> > > You can stand up on flaps, you can stand up inside a crate. You can make it so it requires a MOBILITY_USE and MOBILITY_STAND flags if you wanna prevent stunned or paralyzed people from using it
> > 
> > 
> > There's also the interaction issue for some players. I am afraid this PR isn't a good change at all.
> 
> Interaction issue? Standing up from laying on a couch to PDA then laying down again? How does that relate

I know it would bother some people, but, again, it's more the balance issues.

**Comment by Azarak** [comment]
> 
> 
> > > > > > > This PR creates a lot of balance issues and concerns with IC interaction. While this may sound great in principle, it is not in practice. Sorry.
> > > > > > 
> > > > > > 
> > > > > > Can you give examples?
> > > > > 
> > > > > 
> > > > > You have been stunned but you are not stam-critted, you are able to use your PDA to message. You are laying down in a crate, hiding, you can use your PDA to stealth message. You are under flaps, you can use your PDA to stealth message, no radio required. You are buckled to a bed with your PDA on belt, you can use it to message with no sounds. These are a few, the list goes on. I also dislike that you don't need to put any effort into messaging someone else if you are laying down. Use the radio. PDA's can also be used to call bots, if you are down, you can call bots.
> > > > 
> > > > 
> > > > You can stand up on flaps, you can stand up inside a crate. You can make it so it requires a MOBILITY_USE and MOBILITY_STAND flags if you wanna prevent stunned or paralyzed people from using it
> > > 
> > > 
> > > There's also the interaction issue for some players. I am afraid this PR isn't a good change at all.
> > 
> > 
> > Interaction issue? Standing up from laying on a couch to PDA then laying down again? How does that relate
> 
> I know it would bother some people, but, again, it's more the balance issues.

I dont think so personally.

**Comment by Low3** [comment]
Id suggest adding a message that everyone around can see when someone interacts with PDA (like on vore, where you can see stuff like "Tyrone McMelon taps on their PDA screen"). You can either leave for any time someone interacts with PDA or only when lying down.

**Comment by Azarak** [comment]
> 
> 
> Id suggest adding a message that everyone around can see when someone interacts with PDA (like on vore, where you can see stuff like "Tyrone McMelon taps on their PDA screen"). You can either leave for any time someone interacts with PDA or only when lying down.

I second this. Once I had to figure out my victim was PDA'ing with a mute PDA because then I heard sec talk about them recieving that message on syndi comms

**Comment by Owai-Seek** [comment]
I like the idea of relaying the information that someone has sent a PDA message, even if it's muted, but I wonder if it might get a tad annoying seeing that message pop up in chat over and over again?

Perhaps there could be an icon for if someone has their PDA UI open, such as a tiny cellphone, similar to how there's an overhead for `/me` and `/say`.

This wouldn't solve them doing things in, a crate, or locker for example, but you can't exactly see a person in a crate or locker.

As a sidenote, the new-ish layer shifting ability has a similar thing, where it can't be used when laying down.

**Comment by Low3** [comment]
My only ideas for less spammy version are: message is shown only when person is horizontal, it plays a short beeping/tapping sound or it shows it only in runechat above person's head, but not in chat window.

**Comment by Sir-Lagsalot** [comment]
You could make it so sending a message always makes a sound, like how some cell phone cameras still play a shutter sound even when muted so you can't clandestinely take pictures of people.

**Comment by Owai-Seek** [comment]
Honestly, I don't see people abusing this more than using it for QoL.

**Comment by Gandalf2k15** [comment]
This has been open for a while and I'm sorting out github. I'm not happy with the changes at all, and to be fair, there isn't much to be gained here. The balance concerns are enough to put me off as well. Reopen this with changes that will address the balance issues.

**Pull request closed without merging**